Update Document.ts - Implement stream.update

belyrs opened the pull request.

belyrs committed:
Update Document.ts - Implement stream.update

amarts commented on packages/modules/src/document/Document.ts:
```diff
@@ -370,6 +370,59 @@ export async function fromContent({
   verifyDataStructure(document)
   return document
 }
+export async function updateStream(
```
nitpick: Newline before any new methods/function.

amarts commented:
Good to add usage of `updateStream()` method in demo.ts or any other scripts.

adi-a11y commented:
> Good to add usage of `updateStream()` method in demo.ts or any other scripts.

Added in demo.ts and raised a new PR

smohan-dw commented on packages/modules/src/document/Document.ts:
```diff
@@ -370,6 +370,59 @@ export async function fromContent({
   verifyDataStructure(document)
   return document
 }
+export async function updateStream(
```
ideal to name this as updateFromContent

belyrs replied:
I will change it and make a PR

smohan-dw commented on packages/modules/src/document/Document.ts:
```diff
@@ -370,6 +370,59 @@ export async function fromContent({
   return document
 }
+export async function updateStream(
+  document: IDocument,
```
Isn't it a better approach to pass the existing IDocument and updated IContent as arguments? This will help the modified content to update the hashes, and we can replace the existing document with the new content. Post this, we can update other sections of the document (signature, dates etc) and create the updated document from which the stream details are captured. What the update should not allow is the change of schema. We need to run validation to ensure the new content uses the schema mentioned in the original document.

smohan-dw commented on packages/modules/src/document/Document.ts:
```diff
@@ -370,6 +370,59 @@ export async function fromContent({
 }
+export async function updateStream(
+  document: IDocument,
+  argContent: any,
```
IContent?

smohan-dw replied:
Better to follow the same argument pattern of fromContent

belyrs committed:
change the name from updateStream to updateFromContent

belyrs committed:
Update Document.ts: Used fromSchemaAndContent and fromContent, Need to check the issue with keyUri

This method has an error!

First of all, have used fromSchemaAndContent and fromContent instead of generating the hashes.

issue: `keyUri` is coming up as undefined after the document updation.
 
I have also printed the issuer signature before  passing it to the document and I noticed keyUri field itself is missing.

belyrs committed:
Merge branch 'dhiway:develop' into stream-update

belyrs committed:
Update Document.ts: updated updateFromContent

Printed the destructured objects so that the build dose not fail. Still need to look into the issue mentioned in the previous commit.

belyrs committed:
Update Document.ts
